src/components/Nav/Nav.Props.ts

```diff
@@ -78,17 +78,21 @@ export interface INavLink {
   isExpanded?: boolean;
 
   /**
-   * Any additional properties to apply to the rendered links.
+   * (Optional) Aria label for nav link
    */
-  [propertyName: string]: any;
+  ariaLabel?: string;
 
   /**
-   * (Optional) Aria label for nav link
+   * (Optional) Unique ID for nav link
    */
-  ariaLabel?: string;
+  id?: string;
```

Comment:
OK revert back to key  this is meta info field needed to store the persisted key for each link during operations.